src/host/ledgerio.h

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "consensus/ledgerenclavetypes.h"
+#include "ds/logger.h"
+
+#include <cstdint>
+#include <cstdio>
+#include <errno.h>
+#include <string>
+#include <sys/types.h>
+#include <unistd.h>
+#include <vector>
+
+namespace asynchost
+{
+  class LedgerIO
```

Comment:
Should this be called `LedgerFile` instead and perhaps we should define an interface for any ledger while we're here?

Reply:
@olgavrou why did we split this? why isn't the pbft code using the exist Ledger directly?

Reply:
un-splitting